File: src/main/java/homeWorkSeven/controller/ProductController.cs
namespace homeWorkSeven{
    public class ProductController 
{
    private static ProductService productService = new ProductServiceImpl();

    public static void SaveProduct()
    {
        productService.SaveProduct();
    }
    public static void SaveProduct(Water water)
    {
        productService.SaveProduct(water);
    }

    public static void GetProduct()
    {
        productService.GetProduct();
    }
}

}

File: src/main/java/homeWorkSeven/controller/ProductController.java
package homeWorkSeven.controller;

import homeWorkSeven.model.Product;
import homeWorkSeven.model.Water;
import homeWorkSeven.service.ProductService;
import homeWorkSeven.service.ProductServiceImpl;

public class ProductController <T extends Product>{
    private static ProductService productService = new ProductServiceImpl();
    public static void saveProduct(Water water){
        productService.saveProduct(water);
    }

    public static void getProduct() {
        productService.getProduct();
    }
}
